test/fixtures/real_html/yelp.html

```diff
@@ -103,7 +103,7 @@
 
 
 
-  <link rel="canonical" href="https://www.yelp.com/search?find_loc=San+Francisco%2C+CA">
+  <link rel="canonical" href="http://my-site.com/">
```

Comment:
i'm not expecting these to be modified actually. `"https://www.yelp.com/search?find_loc=San+Francisco%2C+CA"` shouldn't be classified as having the same host as the server (in the tests) so `Url.addLangCode` should be leaving it alone

Reply:
Would such a case be expected: the canonical tag is intentionally set to point to a different domain? I feel that because we left the feature to default to false, we assume when the user turn this feature on, they are certain that all canonical tags are correct and valid (at least is on the same domain).

Reply:
To be safe I've added a check to verify the canonical URL is same host as request before translating it.